File: B62722/Laboratorios/LaboratorioLibreria/Orbis/src/main/java/com/orbis/b62722/domain/Autor.java
package com.orbis.b62722.domain;

public class Autor {

	private int id_autor;
	private String nombre_autor;
	private String apellidos_autor;
	
	public Autor() {

	}

	public int getId_autor() {
		return id_autor;
	}


	public void setId_autor(int id_autor) {
		if(id_autor <= 0) throw new RuntimeException("El id debe ser mayor a 0.");
		this.id_autor = id_autor;
	}


	public String getNombre_autor() {
		return nombre_autor;
	}


	public void setNombre_autor(String nombre_autor) {
		if(nombre_autor == null || nombre_autor.trim().equals("")) throw new RuntimeException("El nombre es requerido");
		this.nombre_autor = nombre_autor;
	}


	public String getApellidos_autor() {
		return apellidos_autor;
	}


	public void setApellidos_autor(String apellidos_autor) {
		if(apellidos_autor == null || apellidos_autor.trim().equals("")) throw new RuntimeException("El apellido es requerido");
		this.apellidos_autor = apellidos_autor;
	}
	
	
	
	
	
}

File: B53953/Laboratorios/Lab4/OrbisB53953.Web/OrbisB53953.Model/Domain/Autor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace OrbisB53953.Web.Domain
{
    public class Autor
    {

        private int idAutor;
        public int IdAutor
        {
            get
            {
                return idAutor;
            }
            set
            {
                if (value <= 0) throw new Exception("El id debe ser positivo.");
                idAutor = value;
            }
        }

        private string nombreAutor;
        public string NombreAutor
        {
            get
            {
                return nombreAutor;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Nombre es requerido.");
                nombreAutor = value;
            }
        }

        private string primerApellidoAutor;
        public string PrimerApellidoAutor
        {
            get
            {
                return primerApellidoAutor;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Primer apellido es requerido.");
                primerApellidoAutor = value;
            }
        }

        private string segundoApellidoAutor;
        public string SegundoApellidoAutor
        {
            get
            {
                return segundoApellidoAutor;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Segundo apellido es requerido.");
                segundoApellidoAutor = value;
            }
        }

        public Autor(int idAutor, string nombreAutor, string primerApellidoAutor, string segundoApellidoAutor)
        {
            IdAutor = idAutor;
            NombreAutor = nombreAutor;
            PrimerApellidoAutor = primerApellidoAutor;
            SegundoApellidoAutor = segundoApellidoAutor;
        }
    }
}
